charts/consul/templates/mesh-gateway-deployment.yaml

```diff
@@ -213,6 +213,8 @@ spec:
             {{- end }}
             {{- if and .Values.externalServers.enabled .Values.externalServers.tlsServerName }}
             -tls-server-name={{.Values.externalServers.tlsServerName }} \
```

Comment:
Should this be `consul-tls-server-name` as well?

Reply:
I think this should be `-tls-server-name`  for `consul-dataplane` and `-consul-tls-server-name` for other commands.
We should probably clear this up though.

Reply:
I just updated this to use tls-server-name everywhere.  I think differentiating for dataplane vs other components is more of an implementation detail, where this setting in every case is meant to set the TLSConfig.